### PR TITLE
Add KB for the issue `volumes with lot of files take a long time to finish mounting`

### PR DESCRIPTION
**Pull request by PhanLe1010**
longhorn/longhorn#2131

### CODE REVIEW

**Review by yasker — COMMENTED** [review]
LGTM. @catherineluse can you help to take a look as well?

File: content/kb/troubleshooting-volume-take-long-time-to-mount.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+---
+
+## Applicable versions
+All Longhorn versions.
+
+## Symptoms
+
+When starting workload pod which uses Longhorn volumes,
+observe in Longhorn UI that the Longhorn volumes are attached quickly
+but it takes a long time for the volumes to finish mounting and workload to be able to start.
```
Comment (catherineluse) [inline content/kb/troubleshooting-volume-take-long-time-to-mount.md:17]: When starting a workload pod that uses Longhorn volumes,
the Longhorn UI shows that the Longhorn volumes are attached quickly,
but it takes a long time for the volumes to finish mounting and for the workload to be able to start.

File: content/kb/troubleshooting-volume-take-long-time-to-mount.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+This issue only happens when Longhorn volumes that have many files/directories and the workload pod sets `fsGroup` in the `securityContext` similar to as below:
```
Comment (catherineluse) [inline content/kb/troubleshooting-volume-take-long-time-to-mount.md:19]: This issue only happens when the Longhorn volumes have many files/directories and `securityContext.fsGroup` is set in the workload pod as shown below:

File: content/kb/troubleshooting-volume-take-long-time-to-mount.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+spec:
+  securityContext:
+    runAsUser: 1000
+    runAsGroup: 3000
+    fsGroup: 2000
+```
+
+## Reason
+
+By default, when seeing `fsGroup` field, each time a volume is mounted, Kubernetes recursively `chown()` and `chmod()` all the files and directories inside the volume.
```
Comment (catherineluse) [inline content/kb/troubleshooting-volume-take-long-time-to-mount.md:30]: By default, when seeing the `fsGroup` field, each time a volume is mounted, Kubernetes recursively calls `chown()` and `chmod()` on all the files and directories inside the volume.

File: content/kb/troubleshooting-volume-take-long-time-to-mount.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  securityContext:
+    runAsUser: 1000
+    runAsGroup: 3000
+    fsGroup: 2000
+    fsGroupChangePolicy: "OnRootMismatch"
+```
+When `fsGroupChangePolicy` is set to `OnRootMismatch`, if the root of the volume already has the correct permissions,
+the recursive permission and ownership change will be skipped.
+It means that if users don't change the `pod.spec.securityContext.fsGroup` between pod's startups,
+K8s will only have to check the permissions and ownership of the root and the mounting process will be much faster compared to always recursively change the volumes' ownership and permissions.
```
Comment (catherineluse) [inline content/kb/troubleshooting-volume-take-long-time-to-mount.md:49]: compared to always recursively changing

**Review by catherineluse — CHANGES_REQUESTED** [review]

File: content/kb/troubleshooting-volume-take-long-time-to-mount.md (hunk 1 above)
Comment (PhanLe1010) [reply content/kb/troubleshooting-volume-take-long-time-to-mount.md]: Thank you. Fixed

File: content/kb/troubleshooting-volume-take-long-time-to-mount.md (hunk 2 above)
Comment (PhanLe1010) [reply content/kb/troubleshooting-volume-take-long-time-to-mount.md]: Thank you. Fixed

File: content/kb/troubleshooting-volume-take-long-time-to-mount.md (hunk 3 above)
Comment (PhanLe1010) [reply content/kb/troubleshooting-volume-take-long-time-to-mount.md]: Thank you. Fixed

File: content/kb/troubleshooting-volume-take-long-time-to-mount.md (hunk 4 above)
Comment (PhanLe1010) [reply content/kb/troubleshooting-volume-take-long-time-to-mount.md]: Thank you. Fixed

**Comment by PhanLe1010** [comment]
@yasker 

Catherine's review is addressed

**Comment by yasker** [comment]
@catherineluse can you review again? Thanks.

**Review by catherineluse — APPROVED** [review]

**Pull request merged**